app/assets/javascripts/angular/controllers/frame_ctrl.js

```diff
@@ -36,33 +36,37 @@ angular.module("Prometheus.controllers").controller('FrameCtrl', ["$scope",
   };
 
   function buildFrameURL(url) {
-    var parser = document.createElement('a');
+    var parser = document.createElement("a");
     parser.href = url;
-    var queryStringComponents = parser.search.substring(1).split('&');
+    var queryStringComponents = parser.search.substring(1).split("&");
     if ($scope.frame.graphite) {
-      if (url.indexOf("until") === -1) {
-        queryStringComponents.push("until=now");
-      }
+      ["height=0", "width=0", "from=-1h", "until=now"].forEach(function(e) {
+        // Unshift elements to the front of the query string. If `from`
+        // already exists, it will not be clobbered by these values.
+        queryStringComponents.unshift(e);
+      });
       queryStringComponents = queryStringComponents.map(function(e) {
         switch (0) {
-        case e.indexOf('height='):
+        case e.indexOf("height="):
           return setDimension(e, $scope.frameHeight().height);
-        case e.indexOf('width='):
+        case e.indexOf("width="):
           var width = $scope.frameHeight().height / $scope.aspectRatio;
           return setDimension(e, width);
-        case e.indexOf('from='):
+        case e.indexOf("from="):
           if (!$scope.frame.range) {
+            // If the frame doesn't specify a time range, use the time range
+            // in the query string.
```

Comment:
This is unexpected. If a dashboard is set to "range 3h" then a graphite frame will only show 1h. I'd vote to reduce the number of branches in the code and to always use the frame range (which should default to the dashboard range if not overridden).